File: compatLib/compatLibVT/src/app/lawnchair/compatlib/thirteen/QuickstepCompatFactoryVT.java
package app.lawnchair.compatlib.thirteen;

import android.app.IApplicationThread;
import android.window.IRemoteTransition;
import android.window.RemoteTransition;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import app.lawnchair.compatlib.ActivityManagerCompat;
import app.lawnchair.compatlib.ActivityOptionsCompat;
import app.lawnchair.compatlib.RemoteTransitionCompat;
import app.lawnchair.compatlib.twelve.QuickstepCompatFactoryVS;

public class QuickstepCompatFactoryVT extends QuickstepCompatFactoryVS {

    @NonNull
    @Override
    public ActivityManagerCompat getActivityManagerCompat() {
        return new ActivityManagerCompatVT();
    }

    @NonNull
    @Override
    public ActivityOptionsCompat getActivityOptionsCompat() {
        return new ActivityOptionsCompatVT();
    }

    @NonNull
    @Override
    public RemoteTransitionCompat getRemoteTransitionCompat() {
        return new RemoteTransitionCompat() {
            @Override
            public RemoteTransition getRemoteTransition(@NonNull IRemoteTransition remoteTransition,
                                                        @Nullable IApplicationThread appThread,
                                                        @Nullable String debugName) {
                return new RemoteTransition(remoteTransition, appThread);
            }
        };
    }
}

File: compatLib/compatLibVT/src/app/lawnchair/compatlib/thirteen/QuickstepCompatFactoryVT.kt
package app.lawnchair.compatlib.thirteen;

import android.app.IApplicationThread;
import android.window.IRemoteTransition;
import android.window.RemoteTransition;

import app.lawnchair.compatlib.ActivityManagerCompat;
import app.lawnchair.compatlib.ActivityOptionsCompat;
import app.lawnchair.compatlib.RemoteTransitionCompat;
import app.lawnchair.compatlib.twelve.QuickstepCompatFactoryVS;

open class QuickstepCompatFactoryVT : QuickstepCompatFactoryVS() {

    override fun getActivityManagerCompat(): ActivityManagerCompat {
        return ActivityManagerCompatVT()
    }

    override fun getActivityOptionsCompat(): ActivityOptionsCompat {
        return ActivityOptionsCompatVT()
    }

    override fun getRemoteTransitionCompat(): RemoteTransitionCompat {
        return object : RemoteTransitionCompat() {
            override fun getRemoteTransition(
                remoteTransition: IRemoteTransition,
                appThread: IApplicationThread?,
                debugName: String?
            ): RemoteTransition {
                return RemoteTransition(remoteTransition, appThread)
            }
        }
    }
}
